### PR TITLE
add logic to get message type

### PR DESCRIPTION
**Pull request by jgrant22**

### CODE REVIEW

File: src/main/java/io/github/linuxforhealth/hl7/parsing/HL7DataExtractor.java [hunk 1]
```diff
@@ -337,12 +337,29 @@ private Terser getTerser() {
...
   public static String getMessageType(Message message) {
-    return message.getName();
+    try {
+      MSH msh = (MSH) message.get("MSH");
+      String theActualMessageType = msh.getMessageType().getMsg1_MessageCode().getValue() + "_" + msh.getMessageType().getMsg2_TriggerEvent().getValue();
+      System.out.println(message.getName());
+      return theActualMessageType;
+    } catch (HL7Exception e){
+      // TODO Auto generated catch block
+      e.printStackTrace();
```
Comment (jgrant22) [inline src/main/java/io/github/linuxforhealth/hl7/parsing/HL7DataExtractor.java:347]: wasn't sure what this meant?  the TODO comment

File: src/test/java/io/github/linuxforhealth/FHIRConverterTest.java [hunk 2]
```diff
@@ -195,6 +195,16 @@ public void test_blank_message_throws_error() throws IOException {
...
+//  @Test
+//  public void test_adt_40_message() {
+//    String hl7message =
+//            "MSH|^~\\&|REGADT|MCM|RSP1P8|MCM|200301051530|SEC|ADT^A40^ADT_A39|00000003|P|2.5\n" +
+//            "EVN|A40|200301051530\n" +
+//            "PID|||MR1^^^XYZ||MAIDENNAME^EVE\n" +
+//            "MRG|MR2^^^XYZ";
+//    HL7ToFHIRConverter ftv = new HL7ToFHIRConverter();
+//    String json = ftv.convert(hl7message, OPTIONS);
+//  }
```
Comment (jgrant22) [inline src/test/java/io/github/linuxforhealth/FHIRConverterTest.java:207]: I was using this test just to debug should I add to it and leave it commented out?

File: src/main/java/io/github/linuxforhealth/hl7/parsing/HL7DataExtractor.java (hunk 1 above)
Comment (LisaWellman) [reply src/main/java/io/github/linuxforhealth/hl7/parsing/HL7DataExtractor.java:347]: It is a default chunk of code generated by the IDE -- need to do something appropriate in the catch block

File: src/main/java/io/github/linuxforhealth/hl7/parsing/HL7DataExtractor.java [hunk 3]
```diff
@@ -337,12 +337,29 @@ private Terser getTerser() {
   }
 
   public static String getMessageType(Message message) {
-    return message.getName();
+    try {
+      MSH msh = (MSH) message.get("MSH");
+      String theActualMessageType = msh.getMessageType().getMsg1_MessageCode().getValue() + "_" + msh.getMessageType().getMsg2_TriggerEvent().getValue();
```
Comment (LisaWellman) [inline src/main/java/io/github/linuxforhealth/hl7/parsing/HL7DataExtractor.java:342]: This was Elena's hack-ish attempt to get the right value.  She was hoping there was a smarter way to get the right value.

File: src/main/java/io/github/linuxforhealth/hl7/parsing/HL7DataExtractor.java [hunk 4]
```diff
@@ -337,12 +337,29 @@ private Terser getTerser() {
...
+      System.out.println(message.getName());
```
Comment (LisaWellman) [inline src/main/java/io/github/linuxforhealth/hl7/parsing/HL7DataExtractor.java:343]: Remove println

File: src/main/java/io/github/linuxforhealth/hl7/parsing/HL7DataExtractor.java [hunk 5]
```diff
@@ -337,12 +337,29 @@ private Terser getTerser() {
...
+      // TODO Auto generated catch block
+      e.printStackTrace();
+    }
+    return null;
   }
 
 
   public String getMessageType() {
-    return message.getName();
+    try {
```
Comment (LisaWellman) [inline src/main/java/io/github/linuxforhealth/hl7/parsing/HL7DataExtractor.java:354]: Please refactor the code so it is not duplicated.  This method should be able to just call the other:
`return getMessageType(message)`

File: src/test/java/io/github/linuxforhealth/FHIRConverterTest.java (hunk 2 above)
Comment (LisaWellman) [reply src/test/java/io/github/linuxforhealth/FHIRConverterTest.java:207]: Change the test so that it checks the message type retuned, and then uncomment.

**Review by LisaWellman — CHANGES_REQUESTED** [review]

File: src/test/java/io/github/linuxforhealth/FHIRConverterTest.java (hunk 2 above)
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/FHIRConverterTest.java:207]: gotcha thanks

File: src/main/java/io/github/linuxforhealth/hl7/parsing/HL7DataExtractor.java (hunk 5 above)
Comment (jgrant22) [reply src/main/java/io/github/linuxforhealth/hl7/parsing/HL7DataExtractor.java:354]: thanks, done

**Review by evbaron — APPROVED** [review]
Changes look good. 
Made sure both static and instance method changed to using MSH Event Type/Trigger.
Unit test present.

**Pull request merged**